src/main/java/nextstep/subway/domain/Line.java

```diff
@@ -1,22 +1,133 @@
 package nextstep.subway.domain;
 
+import org.apache.commons.lang3.StringUtils;
+
 import javax.persistence.*;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
 
 @Entity
 public class Line extends BaseEntity {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
+
     @Column(unique = true)
     private String name;
+
     private String color;
 
-    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
-    private List<Section> sections = new ArrayList<>();
+    @OneToMany(
+            mappedBy = "line",
+            cascade = {CascadeType.PERSIST, CascadeType.MERGE},
+            orphanRemoval = true,
+            fetch = FetchType.LAZY
+    )
+    private List<Section> sections = new LinkedList<>();
+
+    public void init(Section section) {
+        section.setLine(this);
+    }
+
+    public void addSection(Section section) {
+        verifyCanBeAdded(section);
+
+        var stations = getStations();
+        if (stations.get(0).equals(section.getDownStation())) {
+            section.setLine(this);
+            return;
+        }
+
+        if (stations.get(stations.size() - 1).equals(section.getUpStation())) {
+            section.setLine(this);
+            return;
+        }
+
+        addSectionToBetween(section);
+    }
+
+    public List<Station> getStations() {
+        if (sections.isEmpty()) {
+            return new ArrayList<>();
+        }
+
+        var ret = new ArrayList<Station>();
+        var top = firstSection();
```

Comment:
```suggestion
        var top = findFirstSection();
```
```suggestion
        var top = fetchFirstSection();
```
등등 함수명을 작성할때는 동사로 작성해주시면 좋을 것 같습니다 😄